.gitlab/includes/hip_pipeline.yml

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) 2023 ETH Zurich
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file BOOST_LICENSE_1_0.rst or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+include:
+  - local: '.gitlab/includes/common_pipeline.yml'
+  - local: '.gitlab/includes/common_spack_pipeline.yml'
+
+.variables_gcc12_hip5_config:
+  variables:
+    ARCH: linux-ubuntu22.04-zen3
+    BUILD_TYPE: Debug
+    COMPILER: gcc@12.2.0
+    CXXSTD: 20
+    GPU_TARGET: 'gfx90a'
+    SPACK_SPEC: "pika@main arch=$ARCH %${COMPILER} +rocm +stdexec amdgpu_target=${GPU_TARGET} \
+                 malloc=system cxxstd=$CXXSTD ^boost@1.79.0 ^hwloc@2.6.0 ^hip@5.5 ^llvm~gold \
+                 ^fmt@10.0.0 ^stdexec@git.nvhpc-23.09.rc4=main"
+
+gcc12_hip5_spack_image:
+  stage: spack_configs
+  timeout: 4 hours
+  needs: [base_spack_image]
+  extends:
+    - .container-builder
+    - .variables_gcc12_hip5_config
+  before_script:
+    - CONFIG_TAG=`echo $SPACK_SPEC-$SPACK_COMMIT | sha256sum - | head -c 16`
+    - compiler=${COMPILER/@/-}
+    - export PERSIST_IMAGE_NAME=$CSCS_REGISTRY_PATH/pika-$compiler:$CONFIG_TAG
+    - echo -e "compiler=$compiler\nBASE_IMAGE=$PERSIST_IMAGE_NAME" >> compiler.env
+  variables:
+    BASE_IMAGE: $CSCS_REGISTRY_PATH/pika-spack-base:$SPACK_COMMIT
+    DOCKERFILE: .gitlab/docker/Dockerfile.spack_compiler
+    DOCKER_BUILD_ARGS: '["BASE_IMAGE","ARCH","COMPILER","SPACK_SPEC"]'
+  artifacts:
+    reports:
+      dotenv: compiler.env
+
+gcc12_hip5_debug_build:
+  stage: build
+  timeout: 2 hours
+  extends:
+    - .container-builder
+    - .variables_gcc12_hip5_config
+    - .cmake_variables_common
+  needs:
+    - gcc12_hip5_spack_image
+  before_script:
+    - build_type=`echo $BUILD_TYPE | tr '[:upper:]' '[:lower:]'`
+    - configuration=$CMAKE_COMMON_FLAGS-$CMAKE_FLAGS-$CI_COMMIT_SHORT_SHA
+    - configuration=${configuration//-D/}
+    - CONFIG_TAG=`echo $configuration | sha256sum - | head -c 16`
+    - export PERSIST_IMAGE_NAME=$CSCS_REGISTRY_PATH/pika-$compiler-$build_type-build:$CONFIG_TAG
+    - echo -e "PERSIST_IMAGE_NAME=$PERSIST_IMAGE_NAME" > build.env
+  variables:
+    DOCKERFILE: .gitlab/docker/Dockerfile.spack_build
+    CMAKE_FLAGS: "-DCMAKE_BUILD_TYPE=$BUILD_TYPE -DPIKA_WITH_CXX_STANDARD=$CXXSTD \
+                  -DPIKA_WITH_HIP=ON -DPIKA_WITH_MALLOC=system \
+                  -DCMAKE_HIP_ARCHITECTURES=$GPU_TARGET -DPIKA_WITH_STDEXEC=ON"
+    DOCKER_BUILD_ARGS: '["BASE_IMAGE","SOURCE_DIR","BUILD_DIR","CMAKE_COMMON_FLAGS","CMAKE_FLAGS"]'
+  artifacts:
+    reports:
+      dotenv: build.env
+
+gcc12_hip5_debug_test:
+  extends:
+    - .variables_gcc12_hip5_config
+    - .test_common_gpu_clariden_hip
+  needs:
+    - gcc12_hip5_debug_build
+  script:
+    - spack arch
+    - spack build-env $spack_spec -- bash -c "ctest --label-exclude COMPILE_ONLY --test-dir ${BUILD_DIR} -j$(nproc) --timeout 120 --output-on-failure --no-compress-output --no-tests=error"
```

Comment:
Make sure this is in line with other configurations (in particular `-j`). #862 would be helpful here so I would try to bump up the priority of that issue.